app/controllers/declaration/SummaryController.scala

```diff
@@ -82,36 +86,32 @@ class SummaryController @Inject() (
     )
   }
 
-  def displayPageOnMandatoryData(mode: Mode)(implicit request: JourneyRequest[_]): Future[Result] =
-    if (containsMandatoryData(request.cacheModel, mode)) displaySummaryPage
-    else Future.successful(Ok(summaryPageNoData()))
-
-  private def displaySummaryPage(implicit request: JourneyRequest[_]): Future[Result] = {
-
+  private def displaySummaryPage(mode: Mode)(implicit request: JourneyRequest[_]): Future[Result] = {
     val maybeLrn = request.cacheModel.lrn.map(Lrn(_))
     val backlink =
       if (request.cacheModel.parentDeclarationEnhancedStatus.contains(ERRORS)) SubmissionsController.displayListOfSubmissions()
       else SavedDeclarationsController.displayDeclarations()
     val duplicateLrnError = Seq(lrnDuplicateError)
 
     isLrnADuplicate(maybeLrn).map { lrnIsDuplicate =>
-      if (lrnIsDuplicate) Ok(normalSummaryPage(backlink, duplicateLrnError))
-      else Ok(amendDraftSummaryPage(backlink))
+      if (lrnIsDuplicate) Ok(normalSummaryPage(mode, backlink, duplicateLrnError))
+      else Ok(amendDraftSummaryPage(mode, backlink))
     }
   }
 
-  private val hrefSourcePattern = """href="/customs-declare-exports/declaration/.+\?mode=Change"""".r
-  private val hrefDestPattern = s"""href="$continuePlaceholder""""
+  private val hrefSource = """href="/customs-declare-exports/declaration/.+\?mode="""
```

Comment:
This is a temporary solution. When we remove all modes, we'll need to find a different solution to differentiate, in the "Summary" page, rows with a `Change` link from any other link included in the page pointing to  `/customs-declare-exports/declaration/{...}` but that is not part of the Summary table.